src/types.ts

```diff
@@ -377,6 +377,77 @@ type TurboServiceConfiguration = {
   token?: TokenType;
 };
 
+export type TurboUploadEventsAndPayloads = {
+  'upload-progress': {
+    totalBytes: number;
+    processedBytes: number;
+  };
+  'upload-error': Error; // TODO: replace with FailedRequestError
```

Comment:
nit: shipping a lot of TODOs around these error classes

Reply:
[53e5f94](https://github.com/ardriveapp/turbo-sdk/pull/256/commits/53e5f9411bfa9d99e37c652c634234bd271fdf25)